sdk/core/azure-core/src/main/java/com/azure/android/core/internal/util/serializer/JacksonAdapter.java

```diff
@@ -309,4 +302,22 @@ private String serializeRaw(Object object) {
             return null;
         }
     }
+
+    private static String join(@NonNull CharSequence delimiter, @NonNull Iterable tokens) {
+        final Iterator<?> iterator = tokens.iterator();
+
+        if (!iterator.hasNext()) {
+            return "";
+        }
+
+        final StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(iterator.next());
+
+        while (iterator.hasNext()) {
+            stringBuilder.append(delimiter);
+            stringBuilder.append(iterator.next());
+        }
+
+        return stringBuilder.toString();
```

Comment:
We need `stringBuilder.append(iterator.next())` because if not the string will start with a delimiter. Hence why we check if the iterator has a next at the beginning.